python/pyarrow/tests/test_parquet.py

```diff
@@ -2504,12 +2682,21 @@ def _test_write_to_dataset_with_partitions(base_path,
     # partitioned dataset
     dataset = pq.ParquetDataset(base_path,
                                 filesystem=filesystem,
-                                validate_schema=True)
+                                validate_schema=True,
+                                use_legacy_dataset=use_legacy_dataset)
     # ARROW-2209: Ensure the dataset schema also includes the partition columns
-    dataset_cols = set(dataset.schema.to_arrow_schema().names)
+    if use_legacy_dataset:
+        dataset_cols = set(dataset.schema.to_arrow_schema().names)
+    else:
+        # TODO(dataset) schema property is an arrow and not parquet schema
```

Comment:
> It isn't clear to me why this is a problem here, since you're just getting the field names of the dataset's schema. Do you need a property accessor for the parquet schema on parquet.dataset.Dataset or ..Fragment?

It's not necessarily a problem (so I should maybe remove the TODO), it's just a difference in API between the old and new dataset, and a difference I don't want to "fix" (in the new API, we should keep `.schema` as the arrow schema IMO, not parquet schema)

Getting access to the schema can probably be done through exposing the parquet metadata (which has this schema). But I think this can be done in a separate PR.

Reply:
```suggestion
        # NB schema property is an arrow and not parquet schema
```